telethon/client/updates.py

```diff
@@ -211,7 +213,46 @@ async def on_greeting(event):
         """
         return [(callback, event) for event, callback in self._event_builders]
 
-    async def catch_up(self: 'TelegramClient'):
+    async def _catch_up_channel(self: 'TelegramClient', channel_id: int, pts: int, limit: int = None):
+        if self._state_cache[channel_id]:
+            pts = self._state_cache[channel_id]
+
+        if not pts:
+            # First-time, can't get difference. Get pts instead.
+            result = await self(functions.channels.GetFullChannelRequest(channel_id))
+            self._state_cache[channel_id] = result.full_chat.pts
+            self._update_state_for(channel_id)
+            return
+        try:
+            while True:
+                d = await self(functions.updates.GetChannelDifferenceRequest(
+                    channel=channel_id,
+                    filter=types.ChannelMessagesFilterEmpty(),
+                    pts=pts,
+                    limit=limit or 100000  # If the limit isn't set, fetch all updates that we can
```

Comment:
I believe the limit should be a lot less for channels, see https://core.telegram.org/api/updates#recovering-gaps, although https://core.telegram.org/method/updates.getChannelDifference seems to hint something else.

Reply:
Set the default limit to 1000